web/src/stream_settings_ui.js

```diff
@@ -648,7 +652,12 @@ export function setup_page(callback) {
         stream_create.set_up_handlers();
 
         const throttled_redraw_left_panel = _.throttle(redraw_left_panel, 50);
-        $("#stream_filter input[type='text']").on("input", () => {
+        $("#search_stream_name").on("input", () => {
+            if (!$("#search_stream_name").val()) {
+                $("#clear_search_stream_name").hide();
+            } else {
+                $("#clear_search_stream_name").show();
+            }
```

Comment:
I feel like you should put these 5 lines of code into a function `update_clear_search_widget`, and just call that function in the 3-4 places you need, rather than having all all the variations; it'd be a lot easier to read.